lib/sugar_utils/file.rb

```diff
@@ -33,6 +36,27 @@ def self.flock_exclusive(file, options = {})
       Timeout.timeout(timeout) { file.flock(::File::LOCK_EX) }
     end
 
+    # Change all of the access values for the specified file including:
+    # * owner
+    # * group
+    # * permissions
+    #
+    # @param filename [String]
+    # @param owner [nil, Integer, String]
+    # @param group [nil, Integer, String]
+    # @param permission [nil, Integer]
+    #
+    # @raise [SugarUtils::File::Error]
+    #
+    # @return [void]
+    def self.change_access(filename, owner, group, permission)
+      FileUtils.chown(owner, group, filename)
+      FileUtils.chmod(permission, filename) if permission
```

Comment:
Should the permission be an optional parameter if we're not necessarily using it?

Reply:
Maybe. They could all be optional parameters. But I do have a reason that all the parameters are required, but can be nil.

So my justification: .change_access is intended to try and set all of the parameters it can, and if any of them are nil just skip them. I think that this will make the code which calls it simpler. You can see the calls in .touch, .write and .append do not need any extra logic because the just pass in all the options.

Then the FileUtils.chmod needs to be explicitly skipped, because it will not handle a nil.
But FilesUtils.chown will already silently ignore nil, as desired.

What do you think?
Do you think it would be more usable if I added this justification to  the method documentation?